File: abstractmodules/src/impl/com/extensiblejava/calculator/MonthlyPaymentCalculator.java
package com.extensiblejava.calculator;

import java.math.BigDecimal;

class MonthlyPaymentCalculator {

	public BigDecimal calculatePayment(BigDecimal presentValue, BigDecimal rate, int term) {
		double dPresentValue = presentValue.doubleValue();
		double dRate = rate.doubleValue() / (1200);

		double revisedRate = dRate + 1;
		double dTerm = term;
		double powRate = Math.pow(revisedRate, dTerm);

		double left = powRate * dPresentValue;

		double middle = dRate / (powRate - 1);

		double right = 1 / (1);

		BigDecimal payment = new BigDecimal(left * middle * right);
		return payment.setScale(2, BigDecimal.ROUND_UP);

	}
}

File: colocateexceptions/osgi/scala/src/MonthlyPaymentCalculator.scala
package com.extensiblejava.calculator.scala

import scala.math._

class MonthlyPaymentCalculator {
	def calculatePayment(presentValue:BigDecimal, rate:BigDecimal, term:Int): BigDecimal = {		
		val dPresentValue = presentValue.toDouble
		val dRate = rate.toDouble / 1200
		
		val revisedRate = dRate + 1;
		val dTerm = term.toDouble
		
		val powRate = pow(revisedRate, dTerm)
		
		val left = powRate * dPresentValue
		val middle = dRate / (powRate - 1)
		val right = 1/(1);
		
		val payment = BigDecimal(left * middle * right).setScale(2, BigDecimal.RoundingMode.HALF_UP)
		payment
	}
}